src/include/migraphx/heap.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_HEAP_HPP
+#define MIGRAPHX_GUARD_RTGLIB_HEAP_HPP
+
+#include <thread>
+#include <cmath>
+#include <algorithm>
+#include <vector>
+#include <cassert>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+template <class T, class Compare>
+struct heap
+{
+    std::vector<T> data;
+    Compare compare;
+
+    heap(const std::vector<T>& val, Compare comp) : data(val), compare(std::move(comp))
+    {
+        for(int i = 2; i <= data.size(); ++i)
+        {
+            std::push_heap(data.begin(), data.begin() + i, compare);
```

Comment:
Why don't we call `make_heap` here?

Reply:
Previous, I had tried to make the unsorted output to be the same between the ref and gpu, by using the push_heap, we know specific content value in the heap, so we can get the same heap between ref and gpu. That is why I use the push_heap to push element one by one. If we always sort the out, yes, we can use make_heap.

Reply:
Alright, then lets use `make_heap` as I think that will be faster.